### DISCUSSION TITLE
Alias for value types is indispensable

### DISCUSSION THREAD

**Discussion by YT-GameMaker** [dotnet/csharplang#7642] [Language Ideas] [answered]
With static abstract members, it has become possible to write generic math libraries. However, when applying generic math, generics can make the code difficult to understand. Currently, file-scoped aliases can be used to alleviate this problem, but this is not enough for a large project. The following code shows a solution to this problem.
```
public struct Vector3<T> { }

public float3 = Vector3<float>;
public double3 = Vector3<double>;

#if USING_DOUBLE
public real3 = double3;
#else
public real3 = float3;
#endif
```
In this example, `Vector3<float>` has an alias `float3`, these two names represent the same type, and `float3` still exists after the code is compiled, which can be referenced by other projects.
If `Vector3<T>` is a reference type, you can eliminate generics and rename the type through a simple inheritance, but this method does not work for value types.
This example also shows the real type of the alias determined by the compile condition, for easy overall switching of the data precision of the project and the precision of any project that depends on this math library. Without aliases, this is almost impossible to achieve. In the field of game development, this is a necessary feature for creating large open worlds.

**Comment by ufcpp** [comment]
How about `global using`?

**Reply by YT-GameMaker** [reply]
Aliases should have the same status as normal types to support various development needs, such as internal types, private types, and public types.

**Reply by FaustVX** [reply]
What you need is `explicit extension`.

**Reply by YT-GameMaker** [reply]
What is this? I can't find any information about it.

**Reply by TheUnlocked** [reply, marked as answer]
It's a proposed feature. See #5497

**Reply by tannergooding** [reply]
> Aliases should have the same status as normal types to support various development needs, such as internal types, private types, and public types.

Aliases and normal types are explicitly different. One is an alias and is erased, the other is a concretely unique thing and changes the ABI of the places it is used. It allows overloading and other features to then exist. 

`global using` can already be made available to consumers of your NuGet package and other considerations and work as aliases are intended to work.

If you instead wanted a friendly syntax that functioned as a transparent wrapper type, that would be its own unique feature.
